apps/confidential/confidential.go

```diff
@@ -176,10 +185,36 @@ func NewCredFromAssertionCallback(callback func(context.Context, AssertionReques
 	return Credential{assertionCallback: callback}
 }
 
-// NewCredFromCert creates a Credential from an x509.Certificate and a PKCS8 DER encoded private key.
-// CertFromPEM() can be used to get these values from a PEM file storing a PKCS8 private key.
+// NewCredFromCert creates a Credential from an x509.Certificate and an RSA private key.
+// CertFromPEM() can be used to get these values from a PEM file.
 func NewCredFromCert(cert *x509.Certificate, key crypto.PrivateKey) Credential {
-	return Credential{cert: cert, key: key}
+	cred, _ := NewCredFromCertChain([]*x509.Certificate{cert}, key)
```

Comment:
what is the story on the discarded error here?

Reply:
`NewCredFromCertChain` returns an error when the cert and key are mismatched, or the key isn't an RSA key. These are unlikely errors when the arguments come from `CertFromPEM`, but because handling cert chains entails discovering them, I had `NewCredFromCertChain` report them. `NewCredFromCert` can't return an error, so I maintained its current behavior of allowing problems with its arguments to surface as signing or authentication errors.